src/txmempool.cpp

```diff
@@ -594,10 +594,13 @@ void CTxMemPool::removeForBlock(const std::vector<CTransactionRef>& vtx, unsigne
                 staged.insert(mapTx.project<0>(it));
                 continue;
             }
-            auto res = ApplyCustomTx(viewDuplicate, mempoolDuplicate, tx, Params().GetConsensus(), nBlockHeight, 0, false);
+            CCustomCSView workView(viewDuplicate); // we don't need invalid tx to change the view
+            auto res = ApplyCustomTx(workView, mempoolDuplicate, tx, Params().GetConsensus(), nBlockHeight, 0, false);
             if (!res.ok && (res.code & CustomTxErrCodes::Fatal)) {
                 LogPrintf("%s: Remove conflicting custom TX: %s\n", __func__, tx.GetHash().GetHex());
                 staged.insert(mapTx.project<0>(it));
+            } else {
+                workView.Flush(); // apply changes when they are checked
```

Comment:
Inside `ApplyCustomTx` if it is success, it already flushed, why here need to flush again?

Reply:
Yeah, i forgot that `ApplyCustomTx` do it for us, when tx is invalid it does not flush.